File: Chapter6Java/sbur-redis-repo/src/main/java/com/thehecklers/sburredis/PlaneFinderPoller.java
package com.thehecklers.sburredis;

import org.springframework.data.redis.connection.RedisConnectionFactory;
import org.springframework.scheduling.annotation.EnableScheduling;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;
import org.springframework.web.reactive.function.client.WebClient;


@EnableScheduling
@Component
public class PlaneFinderPoller {

    private final WebClient client = WebClient.create("http://localhost:7634/aircraft");

    private final RedisConnectionFactory connectionFactory;
    private final AircraftRepository aircraftRepository;
    // private final RedisOperations<String, Aircraft> redisOperations;

    public PlaneFinderPoller(RedisConnectionFactory connectionFactory,
                             AircraftRepository aircraftRepository) {
                             // RedisOperations<String, Aircraft> redisOperations){
        this.connectionFactory = connectionFactory;
        // this.redisOperations = redisOperations;
        this.aircraftRepository = aircraftRepository;
    }

    @Scheduled(fixedRate = 10000)
    private void pollPlanes(){

        // This part of code remove all the data from the redis server
           connectionFactory.getConnection().serverCommands().flushDb();

        client.get()
                .retrieve()
                .bodyToFlux(Aircraft.class)
                .filter(plane -> !plane.getReg().isEmpty())
                .toStream()
                .forEach(aircraftRepository::save);
/*
                .forEach(ac -> redisOperations.opsForValue().set(ac.getReg(), ac));

        Objects.requireNonNull(redisOperations.opsForValue()
                        .getOperations()
                        .keys("*"))
                .forEach(ac ->
                        System.out.println(redisOperations.opsForValue().getAndPersist(ac)));
*/


        aircraftRepository.findAll().forEach(System.out::println);

        // connectionFactory.getConnection().serverCommands().save();

    }

}

File: Chapter6Java/sbur-mongo/src/main/kotlin/com/thehecklers/sburmongo/PlaneFinderPoller.kt
package com.thehecklers.sburmongo

import org.springframework.scheduling.annotation.EnableScheduling
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component
import org.springframework.web.reactive.function.client.WebClient
import org.springframework.web.reactive.function.client.bodyToFlux

@Component
@EnableScheduling
class PlaneFinderPoller(private val aircraftRepository: AircraftRepository) {

    private val client =
        WebClient.create("http://localhost:7634/aircraft")

    @Scheduled(fixedRate = 1000)
    private fun pollPlanes(){
        // aircraftRepository.deleteAll();

        client.get()
            .retrieve()
            .bodyToFlux<Aircraft>()
            .filter{!it.reg.isNullOrEmpty()}
            .toStream()
            .forEach { aircraftRepository.save(it) }

        println("--- All Aircraft ---")
        aircraftRepository.findAll().forEach { println(it) }
    }
}
